code/modules/mob/living/carbon/human/species/station/station_special_vr.dm

```diff
@@ -56,142 +56,198 @@
-			if(H.nutrition < 200 && H.nutrition > 150)
-				if(prob(0.5)) //A bit of an issue, not too bad.
-					H << "<span class='info'>You feel rather hungry. It might be a good idea to find some some food...</span>"
-
-			else if(H.nutrition <= 150 && H.nutrition > 100)
-				if(prob(0.5)) //Getting closer, should probably eat some food about now...
-					H << "<span class='warning'>You feel like you're going to snap and give in to your hunger soon... It would be for the best to find some [pick("food","prey")] to eat...</span>"
-
-			else if(H.nutrition <= 100) //Should've eaten sooner!
-				if(H.feral == 0)
-					H << "<span class='danger'><big>Something in your mind flips, your instincts taking over, no longer able to fully comprehend your surroundings as survival becomes your primary concern - you must feed, survive, there is nothing else. Hunt. Eat. Hide. Repeat.</big></span>"
-					log_and_message_admins("has gone feral due to hunger.", H)
-					H.feral += 5 //just put them over the threshold by a decent amount for the first chunk.
-					if(H.stat == CONSCIOUS)
-						H.emote("twitch")
-				if(H.feral + H.nutrition < 150) //Feralness increases while this hungry, capped at 50-150 depending on hunger.
-					H.feral += 1
+	//While regenerating
+	if(H.reviving)
+		H.weakened = 5
+		H.canmove = 0
+		H.does_not_breathe = TRUE
+
+	//Cold/pressure effects when not regenerating
+	else
+		var/datum/gas_mixture/environment = H.loc.return_air()
+		var/pressure2 = environment.return_pressure()
+		var/adjusted_pressure2 = H.calculate_affecting_pressure(pressure2)
+
+		//Very low pressure damage
+		if(adjusted_pressure2 <= 20)
+			H.take_overall_damage(brute=LOW_PRESSURE_DAMAGE, used_weapon = "Low Pressure")
+
+		//Cold hurts and gives them pain messages, eventually weakening and paralysing, but doesn't damage or trigger feral.
+		//NB: 'body_temperature' used here is the 'setpoint' species var
+		var/temp_diff = body_temperature - H.bodytemperature
+		if(temp_diff >= 50)
+			H.shock_stage = min(H.shock_stage + (temp_diff/20), 160) // Divided by 20 is the same as previous numbers, but a full scale
+			H.eye_blurry = max(5,H.eye_blurry)
+
+
+/datum/species/xenochimera/proc/handle_feralness(var/mob/living/carbon/human/H)
+
+	//Low-ish nutrition has messages and eventually feral
+	var/hungry = H.nutrition <= 200
+
+	//At 360 nutrition, this is 30 brute/burn, or 18 halloss. Capped at 50 brute/30 halloss - if they take THAT much, no amount of satiation will help them. Also they're fat.
+	var/shock = H.traumatic_shock > min(60, H.nutrition/10)
+
+	//Caffeinated xenochimera can become feral and have special messages
+	var/jittery = H.jitteriness >= 100
+
+	//To reduce distant object references
+	var/feral = H.feral
+
+//Handle feral triggers and pre-feral messages
+	if(!feral && (hungry || shock || jittery))
+
+		// If they're hungry, give nag messages (when not bellied)
+		if(H.nutrition >= 100 && prob(0.5) && !ismob(H.loc))
```

Comment:
the !ismob(H.loc) check won't work for new bellies, 'cause the location of the prey is now the belly object rather than the mob.